### DISCUSSION TITLE
Cannot select unit conversion in HomeAssistant

### DISCUSSION THREAD

**Discussion by Lackmake** [jomjol/AI-on-the-edge-device#3218] [General]
I have a few smartplugs where i can manually select how HA should display the power values, like this:
![image](https://github.com/user-attachments/assets/417af5fe-caa1-464c-b32c-1d148c0884fc)

For the AI-on-the-edge entities however i cannot. The option is completely missing:
![image](https://github.com/user-attachments/assets/b4d3ed5b-4cd1-4376-bd8f-cadc34d6b830)

What i found is that the rate-sensor is missing the "device_class" attribute. That would have to be added here for both "rates", i think "power" would be correct as thats what my smartplugs report:
https://github.com/jomjol/AI-on-the-edge-device/blob/67c3020d7df7771d6aeae06f435677a4604af48b/code/components/jomjol_mqtt/server_mqtt.cpp#L187-L198

But i am not 100% sure if that is the only reason why home assistant doesnt let me choose the unit....i also found this old thread about the "native_unit_of_measurement": https://community.home-assistant.io/t/unit-of-measurement-2022-4/409860/16

**Comment by caco3** [comment]
The tricky part is that you also can use it for something else than energy (eg. water). Because of this, you have to select the **meter type** in the MQTT config:
![grafik](https://github.com/user-attachments/assets/37ef7d93-22af-4ff7-bd6d-f4d9822214b2)

It then should show up correctly in HA.

But I have to note that I only tested it with a water and a gas meter. I don't measure the electricy with AIOTED.

**Comment by Lackmake** [comment]
I set the correct metertype in AIOTED and the values do show up with the correct unit in HA, but the HA-feature to convert it to a different unit on the fly isnt available. 
But youre correct, i didnt think of other than energy-usecases. In that case the device_class would have to be resolved with a variable like its done for the device_class of "value" in line 188

Edit: For both gas and water the device_class for the rates would be "volume_flow_rate" according to: https://developers.home-assistant.io/docs/core/entity/sensor/#available-device-classes

**Comment by caco3** [comment]
We would need to see if we can extend the existing implementation without breaking any of the existing functionality.

Starting point: https://github.com/jomjol/AI-on-the-edge-device/blob/67c3020d7df7771d6aeae06f435677a4604af48b/code/components/jomjol_flowcontroll/ClassFlowMQTT.cpp#L145

Your link interestingly states some different units, eg. `L/min` although until now we only provide `L/h` and nobody missed it....

**Comment by caco3** [comment]
I created an issue for it: https://github.com/jomjol/AI-on-the-edge-device/issues/3333
Please continue the discussions there.
